addon/-private/fields/attr.ts

```diff
@@ -1,5 +1,6 @@
 import { computed } from '@ember/object';
-export default function(type, options = {}) {
+
+export default function(type: string, options: Record<string, unknown> = {}) {
```

Comment:
@tchak can you clarify the meaning of the typing for `options` used here and elsewhere? Where is `Record` defined?

Reply:
`Record` is a built-in from typescript. https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types

Do you think it would be best to use `Dict` from `orbit`?

Reply:
Ah, thanks for the clarification :+1: 

It is really unfortunate that the name overlaps with Orbit's `Record`. I suppose that for the sake of consistency we could continue using `Dict` for now and then decide when to deprecate its usage everywhere and switch over to `Record`. Unfortunately, that will probably mean moving away from Orbit's custom `Record` interface as well...